sanic_routing/group.py

```diff
@@ -0,0 +1,103 @@
+from .exceptions import InvalidUsage, RouteExists
+
+
+class RouteGroup:
+    def __init__(self, *routes) -> None:
+        if len(set(route.parts for route in routes)) > 1:
+            raise InvalidUsage("Cannout group routes with differing paths")
+
+        if any(routes[-1].strict != route.strict for route in routes):
+            raise InvalidUsage(
+                "Cannout group routes with differing strictness"
+            )
+
+        route_list = list(routes)
+        route_list.pop()
+
+        self._routes = routes
+        self._method_index = {
+            method: route
+            for route in route_list
+            for method in route.defined_methods
+        }
+        self.pattern_idx = 0
+
+    def __str__(self):
+        display = (
+            f"path={self.path or self.router.delimiter} len={len(self.routes)}"
+        )
+        return f"<{self.__class__.__name__}: {display}>"
+
+    def __iter__(self):
+        return iter(self.routes)
+
+    def __getitem__(self, key):
+        return self.routes[key]
+
+    def merge(self, group, overwrite: bool = False):
+        _routes = list(self._routes)
+        for other_route in group.routes:
+            for current_route in self:
+                if current_route == other_route:
+                    if not overwrite:
+                        raise RouteExists(
+                            f"Route already registered: {self.raw_path} "
+                            f"[{','.join(self.methods)}]"
+                        )
+                else:
+                    _routes.append(other_route)
+        self._routes = tuple(_routes)
+
+    @property
+    def labels(self):
+        return self[0].labels
+
+    @property
+    def methods(self):
+        return frozenset(
+            [method for route in self for method in route.methods]
+        )
+
+    @property
+    def params(self):
+        return self[0].params
+
+    @property
+    def parts(self):
+        return self[0].parts
+
+    @property
+    def path(self):
+        return self[0].path
```

Comment:
Are these accessors for the first route good design, or required for backwards compatibility / convenience, so that a RouteGroup may appear as a single Route? If there are not particularly many places where this is accessed, I'd rather put the `[0]` on call site and avoid all these accessors.

Reply:
It is not a great pattern, agreed. Was a matter of convenience while prototyping it. I have another solution that I will push.

Basically, there are fields set on the underlying routes (that will be identical, so `[0]` is chosen as default), that make it easier for compat (and prototyping). Ultimately, we will store the values on the `RouteGroup` since it will be much faster to do this than to have a custom getter.